internal/olm/client/client.go

```diff
@@ -204,6 +206,46 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 				})
 				return false, nil
 			}
+			// Check deployment status
+			if ctx.Err().Error() == "DeadlineExceeded" {
+				dep := &appsv1.Deployment{}
+				depName := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[0].Name
```

Comment:
You need to check all deployments in the install strategy.

Reply:
I was wondering, wouldn't there be just 1 deployment present in the strategy? What other deployments can we expect, when an operator developer deploys the operator.

Reply:
I haven't personally seen a CSV with multiple deployments, but it is totally possible, ex. you may have a different deployment for managing your webhook servers. You should be checking each deployment because the operator is defined by its bundle (CSV, CRDs, etc.), not just one deployment.